src/sql/azdata.proposed.d.ts

```diff
@@ -444,6 +444,23 @@ declare module 'azdata' {
 		errorMessage?: string;
 	}
 
+	export interface MsalError {
```

Comment:
Shouldn't you be adding this as a return type for Prompt and otheras well?

Although this seems like a much more generic error, consider naming it something like `ProviderError` so that any provider can use it to return more detailed error info.

Reply:
prompt needs to be a PromptFailedResult because that one has the canceled field present, where else did you mean for me to add this as a return type?

Reply:
Let's continue in the other thread so it's inline with the diff